### ISSUE TITLE
refactor "load" function

### ISSUE THREAD

**Issue by jeremymanning** [jeremymanning/hypertools#12]
# extend the load function to support the following:
- existing named datasets (convert to CSV files, numpy arrays, pickles, or some other convenient format)
- arbitrary URLs (download CSV files, spreadsheets, and other pandas-supported filetypes)
- local files (any pandas-supported filetype)
- obj files
- seaborn datasets (named)
- scipy/scikit-learn datasets (named)
- other datasets we analyzed in the arxiv preprint and/or JMLR paper
- text (txt, rtf, doc) files
- images (jpg, png, gif, etc. that can be loaded with scikit-image)
- nii files (via niilearn)
- youtube videos?  (extract text/images?)
- music (mp3/wav files?)

## also add some cool additional datasets:
- [umap zoo animals](https://github.com/duhaime/umap-zoo/tree/master/data)
- [project gutenberg books](https://www.gutenberg.org/browse/scores/top) (named?)

## the function should return either:
- a pandas dataframe (if data is tabular)
- formatted text
- a list of text and dataframes

in other words, the load function should return an object that can be directly plotted or manipulated with other hypertools functions.

# Other notes:
- the load function's supported datatypes should be organized in modules so that support for new filetypes and/or extensions can be easily added

**Comment by jeremymanning** [comment]
This issue has been migrated to the primary ContextLab repository as part of the HyperTools 1.0 issue audit: https://github.com/ContextLab/hypertools/issues/273

Closing here; please follow the ContextLab issue for updates.
